test/index.test.ts

```diff
@@ -386,3 +387,33 @@ test('type related tests', () => {
   // const a: Record<string, number> = {}
   // a[myArray[1]] = myArray[0]
 })
+
+test('isInstanceOf', () => {
```

Comment:
Seeing these tests, it seems to work pretty much the same as `isType`?
Can you check the tests for `Generic isType`

Can you explain the main benefit between `isInstanceOf` over `isType`?

Reply:
The main benefit of `instanceof` and `isInstanceOf()` is that it **traverses the prototype chain**. This means that `isInstanceOf(new XMLHttpRequest(), EventTarget)` will **return true** whereas the `isType()` will **return false**.

The reason for the string argument is so that you can refer to common well-known classes in a cross-realm way since `instanceof EventTarget` doesn't work across `<iframe>` or `open()` boundaries but `isInstanceOf()` would.

An awesome side-effect is that you can use `isInstanceOf(thing, "OffscreenCanvas")` in Node.js without needing to actually have a class for the `OffscreenCanvas` since you're just doing string comparisons. This means you can write isomorphic JS code that handles a) cross-realm, b) Node.js where the class doesn't exist and c) traverse the prototype chain.